src/main/java/io/r2dbc/pool/ConnectionPoolConfiguration.java

```diff
@@ -84,6 +93,16 @@ String getValidationQuery() {
         return validationQuery;
     }
 
+    @Nullable
+    Duration getMaxCreateConnectionTime() {
```

Comment:
Let's enforce non-null with timeouts. A `Duration.ZERO` timeout is a good indicator for no timeout.

Reply:
converted to the `Duration.ZERO` and non-null.